velox/functions/prestosql/aggregates/MinMaxAggregates.cpp

```diff
@@ -417,12 +424,34 @@ class NonNumericMinMaxAggregateBase : public exec::Aggregate {
       if (decoded.isNullAt(i)) {
         return;
       }
+
+      checkNestedNull(decoded, i, decoded.base()->typeKind(), isIntermediate);
+
       if (!accumulator->hasValue() ||
           compareTest(compare(accumulator, decoded, i))) {
         accumulator->write(baseVector, indices[i], allocator_);
       }
     });
   }
+
+  FOLLY_ALWAYS_INLINE static void checkNestedNull(
+      const DecodedVector& decoded,
+      vector_size_t index,
+      TypeKind typeKind,
+      bool isIntermediate = false) {
+    switch (typeKind) {
+      case TypeKind::ARRAY:
+      case TypeKind::MAP:
+      case TypeKind::ROW:
+        VELOX_USER_CHECK(
+            isIntermediate || decoded.isNullAt(index) ||
```

Comment:
It seems that all callers of checkNestedNull first check decoded.isNullAt(index) and skip calling checkNestedNull. Therefore, this check here is redundant.

We seem to be switching on typeKind and checking isIntermediate for each row. Can we instead add a bool member variable to tell whether to check for nested nulls or not. It would be nice to set this variable once in the constructor.

Reply:
Fixed by making a check null flag in the registration function and adding this flag to the constructor. It is more elegant now, thanks.